packages/templates/nextjs-marketing-demo/src/components/layout/Header.tsx

```diff
@@ -0,0 +1,38 @@
+'use client';
+
+import React, { useState } from 'react';
+import { Button, Menu } from 'antd';
+import styles from './header.module.css';
+
+const items = [
+  { key: 'home-page', label: 'Home' },
+  { key: 'about', label: 'About' },
+  { key: 'contact', label: 'Contact' },
+  { key: 'careers', label: 'Careers' },
+];
+
+const Header: React.FC = () => {
+  const [current, setCurrent] = useState('home-page');
+
+  return (
+    <div className={styles.header}>
+      <div className={styles.logo}>
+        <span>Logo</span>
+      </div>
+      <div className={styles.nav}>
+        <Menu
```

Comment:
I can see theres some interactivity here, might be good to put Menu into its own component and mark that one as client so the Header itself can remain a RSC.

Reply:
Good call, I'll separate the Menu into its own component.